pkg/network/tracer/offsetguess/tracer.go

```diff
@@ -583,7 +569,7 @@ func (t *tracerOffsetGuesser) checkAndUpdateCurrentOffset(mp *ebpf.Map, expected
 		t.status.Offset_rtt, _ = skipOverlaps(t.status.Offset_rtt, t.sockRanges())
 		t.status.Offset_rtt_var = t.status.Offset_rtt + 4
 	case GuessSocketSK:
-		if t.status.Sport_via_sk == htons(expected.sport) && t.status.Dport_via_sk == htons(expected.dport) {
```

Comment:
why don't we need this anymore?

Reply:
The `skc_num` field is in host order, not in network order like the `inet_sport` field. See https://elixir.bootlin.com/linux/latest/source/include/net/sock.h#L180 and https://elixir.bootlin.com/linux/latest/source/include/net/inet_sock.h#L224; the latter is declared as `__be16`, which is an annotation to specify the value stored there is going to big endian -- network order.